backend/coins/eth/account.go

```diff
@@ -379,23 +379,44 @@ func (account *Account) update() error {
 		}
 	}
 
+	var balance *big.Int
 	if account.coin.erc20Token != nil {
-		balance, err := account.coin.client.ERC20Balance(account.address.Address, account.coin.erc20Token)
-		if err != nil {
-			return errp.WithStack(err)
-		}
-		account.balance = coin.NewAmount(balance)
+		balance, err = account.coin.client.ERC20Balance(account.address.Address, account.coin.erc20Token)
 	} else {
-		balance, err := account.coin.client.Balance(context.TODO(), account.address.Address)
-		if err != nil {
-			return errp.WithStack(err)
-		}
-		account.balance = coin.NewAmount(balance)
+		balance, err = account.coin.client.Balance(context.TODO(), account.address.Address)
+	}
+
+	if err != nil {
```

Comment:
nit: i'd keep this error check duplicated above. it's conventional to have `if err != nil ...` right below the line that defines the error and makes reading it simpler (even though it is more verbose).